File: src/CSM.Application/Common/Security/AppPolicies.cs
using CSM.Domain.Constants;

namespace CSM.Application.Common.Security;

public static class AppPolicies
{
    public const string UserManagement = "UserManagement";
    public const string SiteManagement = "SiteManagement";
    public const string HRManagement = "HRManagement";
    public const string FinanceManagement = "FinanceManagement";
    public const string ProcurementManagement = "ProcurementManagement";
    public const string CompanyManagement = "CompanyManagement";
    public const string ProjectManagement = "ProjectManagement";

    public static readonly string[] ProjectManagementRoles =
    [
        AppRoles.SuperAdmin,
        AppRoles.CompanyAdmin,
        AppRoles.SiteManager,
        AppRoles.Supervisor
    ];

    public static readonly string[] CompanyManagementRoles =
    [
        AppRoles.SuperAdmin
    ];

    public static readonly string[] UserManagementRoles =
    [
        AppRoles.SuperAdmin,
        AppRoles.CompanyAdmin
    ];

    public static readonly string[] SiteManagementRoles =
    [
        AppRoles.SuperAdmin,
        AppRoles.CompanyAdmin,
        AppRoles.SiteManager,
        AppRoles.Supervisor
    ];

    public static readonly string[] HRManagementRoles =
    [
        AppRoles.SuperAdmin,
        AppRoles.CompanyAdmin,
        AppRoles.HROfficer
    ];

    public static readonly string[] FinanceManagementRoles =
    [
        AppRoles.SuperAdmin,
        AppRoles.CompanyAdmin,
        AppRoles.Accountant
    ];

    public static readonly string[] ProcurementManagementRoles =
    [
        AppRoles.SuperAdmin,
        AppRoles.CompanyAdmin,
        AppRoles.SiteManager
    ];
}

File: mobile/csm_app/lib/core/auth/app_policies.dart
import 'app_roles.dart';

class AppPolicies {
  AppPolicies._();

  static const String userManagement = 'UserManagement';
  static const String siteManagement = 'SiteManagement';
  static const String hrManagement = 'HRManagement';
  static const String financeManagement = 'FinanceManagement';
  static const String procurementManagement = 'ProcurementManagement';
  static const String companyManagement = 'CompanyManagement';
  static const String projectManagement = 'ProjectManagement';

  static const List<String> projectManagementRoles = [
    AppRoles.superAdmin,
    AppRoles.companyAdmin,
    AppRoles.siteManager,
    AppRoles.supervisor,
  ];

  static const List<String> companyManagementRoles = [
    AppRoles.superAdmin,
  ];

  static const List<String> userManagementRoles = [
    AppRoles.superAdmin,
    AppRoles.companyAdmin,
  ];

  static const List<String> siteManagementRoles = [
    AppRoles.superAdmin,
    AppRoles.companyAdmin,
    AppRoles.siteManager,
    AppRoles.supervisor,
  ];

  static const List<String> hrManagementRoles = [
    AppRoles.superAdmin,
    AppRoles.companyAdmin,
    AppRoles.hrOfficer,
  ];

  static const List<String> financeManagementRoles = [
    AppRoles.superAdmin,
    AppRoles.companyAdmin,
    AppRoles.accountant,
  ];

  static const List<String> procurementManagementRoles = [
    AppRoles.superAdmin,
    AppRoles.companyAdmin,
    AppRoles.siteManager,
  ];
}
